tests/test_cli.py

```diff
@@ -138,3 +144,63 @@ def test_cli_progress_bar(tmp_empty):
 )
 def test_parse_cli_notebook_parameters(params, expected):
     assert cli._parse_cli_notebook_parameters(params) == expected
+
+
+@pytest.mark.parametrize(
+    "saved_path, exception_msg, exception_type",
+    [
+        (
+            "./abc",
+            "Invalid save_profiling_data, path must be ended with .csv",
+            ValueError,
+        ),
+        (
+            "./abc.py",
+            "Invalid save_profiling_data, path must be ended with .csv",
+            ValueError,
+        ),
+        (
+            "./abc.txt",
+            "Invalid save_profiling_data, path must be ended with .csv",
+            ValueError,
+        ),
+        (
+            "./abc.png",
+            "Invalid save_profiling_data, path must be ended with .csv",
+            ValueError,
+        ),
+        (
+            float(123.0),
```

Comment:
im confused about these and the  values below. we're testing a CLI, so all the inputs must be serialized to strings. how come we're passing floats, dictionaries and tuples?

Reply:
In the jupyter user might execute something like this:
```
_ = execute_notebook(
    "notebook.ipynb", "output.ipynb", profile_memory=True, save_profiling_data=float(123.0))

```

Reply:
If execute execute_notebook directly, it's possible we pass non-string

Reply:
yes, I understand. but this is testing the CLI. either this is a problem with `runner.invoke` or the values are being serialized to strings anyway. 

i think it's best to parameterized this test over different strings then. because the other values are not representative of a real CLI use case.